client/src/components/sign_in.js

```diff
@@ -1,11 +1,34 @@
 import React, { Component } from "react";
+import axios from "axios";
 import { getItems } from "../actions/index";
 
 class SignIn extends Component {
+  constructor() {
+    super();
+    this.state = {
```

Comment:
@FarokhC  all constructor calls need to have a props element with a super call i.e

constructor(props){
   super(props);
}

@vikr01 's I haven't seen that before, is it still possible to grab props from a parent component?

Reply:
If you want to grab props from the parent, you should use `this.props` in the render function.

If your state makes some changes based off props, you should consider using [`static getDerivedStateFromProps`](https://reactjs.org/docs/react-component#static-getderivedstatefromprops). Although, [you probably won't need that](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state).

Reply:
Actually you do have `@babel/plugin-proposal-class-properties` since your functions are being defined as `fn = ()=>...` instead of `fn(){...`. So it's safe to define your `state` like that.